frontend/packages/dev-console/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -93,6 +95,13 @@ const removeEmptyDefaultParams = (task: PipelineTask): PipelineTask => {
   return task;
 };
 
+export const removeEmptyDefaultValueFromParameter = (parameters: PipelineParam[]) =>
+  parameters.map((parameter) =>
+    _.isEmpty(parameter.default)
+      ? { name: parameter.name, description: parameter.description }
+      : parameter,
+  );
```

Comment:
```suggestion
 _.map(parameters, (parameter) =>_.omit(parameter, _.isEmpty(parameter.default) ? ['default'] : []));
```
Just a suggestion, what you have done is also correct.

Reply:
@divyanshiGupta  The issue with using `_.omit` is , when this util is used in `convertBuilderFormToPipeline` , `params` throws a type error saying  : Type 'Partial<PipelineParam>[]' is not assignable to type 'PipelineParam[]'.
  Type 'Partial<PipelineParam>' is not assignable to type 'PipelineParam'.
    Property 'name' is optional in type 'Partial<PipelineParam>' but required in type 'PipelineParam'. 
This is the reason why I had two different implementations earlier which I then updated to use the one without the `omit` in both places. Let me know your thoughts

Reply:
To avoid this you can simply do this:
```
_.map(parameters, (parameter) =>_.omit(parameter, _.isEmpty(parameter.default) ? ['default'] : []) as PipelineParam);
```

But @debsmita1 I just checked the utils.ts file and there is already a function `removeEmptyDefaultParams` (that removes empty default value from task params) above the function that you have created. You can make this function generic and use it wherever we need to remove empty default params.

Reply:
the util `removeEmptyDefaultParams` removes a name-value pair from task params if the value is empty
but the util that I created removes the `default` property form pipeline parameters if it is empty & keeps the name  & description

Reply:
Okay, then just change the name of your fxn to ` removeEmptyDefaultFromPipelineParams` to avoid any confusions.